### PR TITLE
Add Ceph RBD provisioner

### PR DESCRIPTION
**Pull request by cofyc**
This PR is for #200 #99 .

Essentially, this RBD provisioner implementation is a copy of Kubernetes in-tree RBD provisioner. But for integrating with this project, I made some adjustments and rewrote some parts of code.

cc @rootfs

### CODE REVIEW

File: ceph/rbd/claim.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: claim1
+  annotations:
+    volume.beta.kubernetes.io/storage-class: "rbd"
+spec:
+  accessModes:
+    - ReadWriteMany
```
Comment (weiwei04) [inline ceph/rbd/claim.yaml]: not supported by rbd, ReadWriteOnce would be better.

File: ceph/rbd/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: cephfs-provisioner
```
Comment (weiwei04) [inline ceph/rbd/deployment.yaml]: cephfs-provisioner -> ceph-rbd-provisioner

**Comment by cofyc** [comment]
@rootfs Any comments?

File: ceph/rbd/local-start.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+PATH=${PATH}:`pwd`
+
+if [[ "$KUBECONFIG" == "" ]]; then
+	KUBECONFIG=/root/.kube/config
+fi
+
+rbd-provisioner -master=http://127.0.0.1:8080 -kubeconfig=${KUBECONFIG} -logtostderr
```
Comment (rootfs) [inline ceph/rbd/local-start.sh]: add `-id=rbd-provisioner-1` to match readme

File: ceph/rbd/pkg/provision/provision.go [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+				v1.ResourceName(v1.ResourceStorage): resource.MustParse(fmt.Sprintf("%dMi", sizeMB)),
+			},
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				RBD: rbd,
+			},
+		},
+	}
+	// use default access modes if missing
+	if len(pv.Spec.AccessModes) == 0 {
+		pv.Spec.AccessModes = p.getAccessModes()
```
Comment (rootfs) [inline ceph/rbd/pkg/provision/provision.go]: log a warning if this happens

**Comment by cofyc** [comment]
@rootfs Updated, and the conflict is resolved.

**Comment by kfox1111** [comment]
@cofyc I've started a helm chart to make it easy to launch this, and the cephfs provisioner:
https://github.com/kfox1111/charts/tree/ceph-provisioner/stable/ceph-provisioner

The remaining part is to figure out how the command line and volumes need to work. Will the container work with a service account out of the box? will the --id flag use the hostname by default if it is not specified?

**Comment by cofyc** [comment]
@kfox1111 I try to change `identity` default behavior in [this PR](https://github.com/kubernetes-incubator/external-storage/pull/267), then it will not to be specified in general deployment.

**Pull request merged**